napari/_qt/menus/file_menu.py

```diff
@@ -199,7 +199,20 @@ def _rebuild_samples_menu(self):
         ):
             multiprovider = len(samples) > 1
             if multiprovider:
-                menu = self.open_sample_menu.addMenu(plugin_name)
+                # use display_name for the menu item if npe2
+                if plugin_name in [
+                    manifest.name for manifest in _npe2.iter_manifests()
```

Comment:
I would love to not iterate the manifests here. Can we `try/except` `pm.get_manifest(plugin_name)` instead? Especially as we move towards making npe2 shim default and removing npe1 code, the likelihood of that `try` failing will keep getting slimmer.

Reply:
thanks @DragaDoncila 
This was actually what I did first, but then changed it 😬 
What should the exception be?

Reply:
@psobolewskiPhD i believe `get_manifest` throws a `KeyError` for an invalid plugin name so the rough construction should be:

```
try:
    display_name = pm.get_manifest(plugin_name)
except KeyError:
    display_name = plugin_name
# add sample menu item here in one call
```

Let me know if I can expand, I'm on phone so code blocks are hard 😂

Reply:
Thanks, that was perfect. I tested `get_manifest` and indeed it's a keyerror.
Hopefully I have everything correct now, but I don't have a npe1 plugin that has multiple samples that I can think of...